src/radio_instance.c

```diff
@@ -202,6 +248,42 @@ radio_instance_resp_quark(
     return q;
 }
 
+static
+int
+radio_instance_priority_index(
+    RADIO_INSTANCE_PRIORITY priority)
+{
+    if (priority < RADIO_INSTANCE_PRIORITY_LOWEST) {
+        return 0;
+    } else if (priority > RADIO_INSTANCE_PRIORITY_HIGHEST) {
+        return RADIO_INSTANCE_PRIORITY_COUNT - 1;
+    } else {
+        return priority - RADIO_INSTANCE_PRIORITY_LOWEST;
+    }
+}
+
+static
+void
+radio_instance_notify_request_observers(
+    RadioInstance* self,
+    RADIO_REQ code,
+    GBinderLocalRequest* args)
+{
+    GQuark quark = 0;
```

Comment:
Is this meant to be static?

Reply:
No, the quark value depends on the request code, it's fetched on every call if any request observers were ever registered, but no more than once even if there are several request observers with different priorities (and there may be two of them when logging is on, one for the request name/serial and one for the transaction buffer dump, those have different priorities to make sure that hexdump always comes after the general request info). Another place that may smell like over-optimization.

Reply:
Erm, yeah, sorry. I somehow managed to miss the loop. Nevermind.